### PR TITLE
Keep query params when filtering

### PR DESCRIPTION
**Pull request by rcantin-w**
## Who is this for?
Users of search

## What is it doing for them?
Closes https://github.com/wellcomecollection/wellcomecollection.org/issues/9595

When filtering, only the form values are added to the resulting URL's query params. This means that filters that are available but not displayed in the filters options (such as `partOf`, see ticket for how to access) get erased from it.

This now renders out the "hidden" filters but makes them visually hidden, that way they are considered part of the form and get submitted every time.

I also realised we'd not included `partOf.title` in the redirects from the old search. It's the only missing on I could find.

I added a playwright test and a redirect jest test.

### CODE REVIEW

**Comment by alexwlchan** [comment]
Can we add a Playwright test for this?

**Comment by rcantin-w** [comment]
>Can we add a Playwright test for this?

@alexwlchan this was an excellent suggestion as it made me realise this did not work.
I also realised `partOf.title` was stripped in redirections from `/works` -> `/search/works`.

More complex fix coming through + tests.

File: catalogue/webapp/components/SearchFilters/SearchFilters.Desktop.DynamicFilters.tsx [hunk 1]
```diff
@@ -37,7 +37,18 @@ const DynamicFilterArray = ({
   };
 
   const renderDynamicFilter = (f: Filter, i: number, arr: Filter[]) => {
-    return (
+    // We need to have the excluded filters still in the form so their values gets retained in the URL
+    // when more filtering is done (e.g. partOf.title)
+    // Only checkbox types are excluded at the moment
```
Comment (rcantin-w) [inline catalogue/webapp/components/SearchFilters/SearchFilters.Desktop.DynamicFilters.tsx:42]: (the other types are date range and colour which are very specific to one filter each.)

**Comment by rcantin-w** [comment]
@alexwlchan if you have time tomorrow 🙏 🙌

**Comment by alexwlchan** [comment]
This doesn't seem to be working for me; if I check out this branch and run the app locally, the repro steps in the original ticket still cause the issue for me:

1. Go to http://localhost:3000/works/b5cbgunk
2. Click on "VLEBooks", be taken to http://localhost:3000/search/works?partOf.title=VLeBooks
3. Select the Location filter, click "Online"
4. This takes me to http://localhost:3000/search/works?availabilities=online, which loses the partOf.title filter

**Review by alexwlchan — APPROVED** [review]

**Pull request merged**